src/main/java/jenkins/plugins/hipchat/workflow/HipChatStep.java

```diff
@@ -0,0 +1,133 @@
+        this.message = message;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(HipChatStepExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "hipchat";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Publish HipChat Message";
+        }
+
+        public ListBoxModel doFillColorItems(){
+            ListBoxModel listBoxModel = new ListBoxModel();
+            for(Color color : Color.values()) {
+                listBoxModel.add(new ListBoxModel.Option(color.toString(), color.toString(), false));
+            }
+
+           return listBoxModel;
+        }
+
+    }
+
+    public static class HipChatStepExecution extends AbstractSynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L;
+
+        @Inject
+        private transient HipChatStep step;
+        @StepContextParameter
+        transient TaskListener listener;
+
+
+        @Override
+        protected Void run() throws Exception {
+
+            if (StringUtils.isBlank(step.message)) {
+                throw new AbortException("HipChat message not sent. Messasge property must be supplied.");
+            }
+
+            //default to global config values if not set in step, but allow step to override all global settings
+            HipChatNotifier.DescriptorImpl hipChatDesc = (HipChatNotifier.DescriptorImpl) Jenkins.getInstance().getDescriptor(HipChatNotifier.class);
+            String token = step.token != null ? step.token : hipChatDesc.getToken();
+            String room = step.room != null ? step.room : hipChatDesc.getRoom();
+            String server = step.server != null ? step.server : hipChatDesc.getServer();
+            String sendAs = step.sendAs != null ? step.sendAs : hipChatDesc.getSendAs();
+            //default to gray if not set in step
+            String color = step.color != null ? step.color : Color.GRAY.toString();
+            boolean v2enabled = step.v2enabled != null ? step.v2enabled : hipChatDesc.isV2Enabled();
+
+            //not very elegant, but getHipChatService is private in HipChatNotifier
+            HipChatService hipChatService;
+            if(v2enabled) {
+                hipChatService = new HipChatV2Service(server, token, room);
+            } else {
+                hipChatService = new HipChatV1Service(server, token, room, sendAs);
+            }
+
+            listener.getLogger().println(Messages.WorkflowStepConfig(step.server == null, step.token == null, step.room == null, step.color == null));
+
+            logger.log(Level.WARNING, "HipChat publish settings: api v2 - " + v2enabled + " server - " +
+                    server + " token - " + token + " room - " + room);
+
+            hipChatService.publish(step.message, color, step.notify);
```

Comment:
It may be handy to have variable resolving here, just like https://github.com/jenkinsci/hipchat-plugin/blob/master/src/main/java/jenkins/plugins/hipchat/model/NotificationType.java#L89. That requires modifying BuildUtils to work with Run instances.